### ISSUE TITLE
Tuples mess up inlay parameter hints

### ISSUE THREAD

**Issue by olivercoad** [ionide/ionide-vscode-fsharp#1714]
### Describe the bug
Inlay hints treat tuple arguments as separate parameters to a function, leading to the following parameters being labelled wrong.

### Steps to reproduce
Set `"editor.inlayHints.enabled": "on"` in vscode settings to view inlay hints

```fsharp
let inlayHintsTest firstParam tupleParam lastParam = ()
inlayHintsTest "firstParam" ("t1", "t2") "lastParam"
```

### Expected behaviour
The tuple `("t1", "t2")` should be labelled as a whole as `tupleParam` and `"lastParam"` should be labelled correctly.

### Screenshots
![image](https://user-images.githubusercontent.com/24837994/170264200-9e3b2a54-f0d1-426e-9eb6-ee7890eaf9b1.png)

## Machine info
 - OS: manjaro linux-x64
 - .NET SDK version: 6.0.300
 - Ionide version: 6.0.4

### Additional context
I wanted to try out inlay hints now that vscode has added `offUnlessPressed`. It doesn't seem to work though as they aren't visible unless it's set to on. They don't even show up with `onUnlessPressed`. I'm guessing this is probably a vscode issue rather than an ionide issue though.

**Comment by olivercoad** [comment]
nvm about the `offUnlessPressed` bit not working. Just realised I still had `"FSharp.inlayHints.enabled": false,` in my global vscode settings, which I'd forgotten about because it seems to not disable them when `editor.inlayHints.enabled` is `"on"` or unset. But I guess it does still disable them with other values of `editor.InlayHints.enabled`. After removing `FSharp.inlayHints.enabled` from my global settings, `offUnlessPressed` works great. And the hover tooltip to change the setting is nice.

tldr: don't set `FSharp.inlayHints.enabled` anymore, use `editor.inlayHints.enabled` instead.

**Comment by Booksbaum** [comment]
> can you cover this case with testing in your FSAC branch with all the inlayHint fixes?

It's unfortunately not a trivial case: Tuple itself might be named (like here in issue) -> name for whole tuple. Or Items might be named -> name for items.

And in [code](https://github.com/fsharp/FsAutoComplete/blob/2a517fff6c57b5bc3a856b59055855957358e842/src/FsAutoComplete.Core/InlayHints.fs#L356) we currently collect flattened args (-> no distinction between normal curried or tupled param)


Further complication: what to do when tuple items are named, but tuple value is provided as param?:  
![ParamHintTuple](https://user-images.githubusercontent.com/15612932/170337285-309a1994-b4c7-4984-b2a4-4ddc5c062886.png)  
(last line: `f2 t`)  
(Probably either show no param hint or a combined param hint (`(val1,val2)=` or similar))


All doable (`func.CurriedParameterGroups` has that info for params, and AstTraversal can be adjusted to detect tupled vs normal) -- but that requires some work.

<br/>
<br/>
I'm not doing this for fsharp/FsAutoComplete#943.  

But there are some additional enhancements for inlay hints I want to try afterwards. I can attempt to fix this then.
